mayday/mayday.go

```diff
@@ -1,20 +1,42 @@
 package mayday
 
 import (
-	"fmt"
 	"log"
-	"os"
-	"path"
 )
 
-func maybeCreateLink(src, dst, workspace string) error {
-	if src == "" {
-		return nil
+func Run(t Tar, files []File, commands []Command, journals []Journal) error {
+	defer t.Close()
+
+	// copy files and run commands
+	for i := 0; i < len(files); i++ {
+		log.Printf("Collecting file: %q\n", files[i].Name)
+		t.Add(files[i].Content, files[i].Header)
 	}
-	lp := path.Join(workspace, src)
-	log.Printf("Creating link (%s -> %s)", lp, dst)
-	if err := os.Symlink(dst, lp); err != nil {
-		return fmt.Errorf("error creating link: %v", err)
+
+	for i := 0; i < len(commands); i++ {
+		err := commands[i].Run()
+		if err != nil {
+			log.Print(err)
+		}
+		t.Add(commands[i].Content, commands[i].header())
 	}
+
+	for i := 0; i < len(journals); i++ {
+		err := journals[i].Get()
+		if err != nil {
+			log.Print(err)
+		}
+		t.Add(journals[i].Content, journals[i].header())
```

Comment:
The similarity of these loops makes me think another interface might be interesting.  Maybe add should take some sort of Tarable interface that has header() and content() methods or something like that. 

Or maybe this whole function should take a list of things that implement some sort of MaydayCollector interface, which files, commands, and journals could implement.

Or maybe not.  I should go to bed but we should talk more tomorrow.